iothub_client/src/iothubtransport_amqp_common.c

```diff
@@ -157,6 +157,7 @@ typedef struct AMQP_TRANSPORT_DEVICE_TWIN_CONTEXT_TAG
     uint32_t item_id;
     TRANSPORT_CALLBACKS_INFO transport_callbacks;
     void* transport_ctx;
+    AMQP_TRANSPORT_DEVICE_INSTANCE* device;
```

Comment:
Do we have this problem in the DPS client? Does it need to be fixed there (AFAIK we don't retry so I think the answer may be no but worth checking).